src/hooks/proxy.js

```diff
@@ -25,27 +26,25 @@ exports.proxyHook = async function (options) {
         const parsedProxy = new URL(proxyUrl);
 
         validateProxyProtocol(parsedProxy.protocol);
-        const agents = await getAgents(parsedProxy, options.https.rejectUnauthorized);
+        options.agent = await getAgents(parsedProxy, options.https.rejectUnauthorized);
```

Comment:
I'm a bit torn whether this improves readability. I know it's shorter this way, but it made more sense to me when both the options were next to each other and not separated by the big comment. I don't have a strong opinion about this, but would like to learn why you think it's better this way.

Reply:
It's required. Otherwise it would fail if the user provided their own agents.

Reply:
It's because `http2.request` is used here.

Reply:
Maybe I'm blind, but I don't see any difference in behavior between:

```js
const agents = await getAgents(parsedProxy, options.https.rejectUnauthorized);
if (resolvedRequestProtocol === 'http2') {
            options.agent = agents[resolvedRequestProtocol];
} else {
            options.agent = agents;
}
```

and

```js
options.agent = await getAgents(parsedProxy, options.https.rejectUnauthorized);
if (resolvedRequestProtocol === 'http2') {
        options.agent = options.agent[resolvedRequestProtocol];
}
```

🤷‍♂️

Reply:
The `if (resolvedRequestProtocol === 'http2') {` is outside `if (proxyUrl) {`, previously it was inside. You're not blind, it just may not be obvious at first sight. Let me add appropriate comments in this regards :)

Reply:
https://github.com/apify/got-scraping/blob/8ead6d39c0668b355cacd3a4c3a7d267045805ca/src/hooks/proxy.js#L46-L51

Unless you mean something else 🤔